dune_client/api/extensions.py

```diff
@@ -286,3 +333,33 @@ def _refresh(
             raise QueryFailed(f"Error data: {status.error}")
 
         return job_id
+
+    def _fetch_entire_result(
+        self,
+        results: ResultsResponse,
+    ) -> ResultsResponse:
+        """
+        Retrieve the entire results using the paginated API
+        """
+        next_uri = results.next_uri
+        while next_uri is not None:
+            batch = self.get_execution_results_by_url(url=next_uri)
+            results += batch
```

Comment:
the only doubt I have is this operation causes any N^2 behaviour on array expansion to fit the new data.
Should we have an explict:
- results.GrowForSize(results.metadata.total_row_count, results.metadata.total_result_set_size) ?

I know I over think this sometimes.. I'm just uncertain about the time and space complexity of those:
- arrayList + arrayList of rows
- BytesIO.write(bytesIO.read())
 operations

Reply:
on the CSV, I agree w/ you, the only alternative is to let the customer paginate at a higher level in the application stack.